spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-local/src/main/java/org/springframework/cloud/data/module/deployer/local/LocalModuleDeployer.java

```diff
@@ -77,9 +79,11 @@ public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
 		catch (Exception e) {
 			throw new IllegalStateException("failed to determine URL for module: " + module, e);
 		}
-		args.add(String.format("--%s.server.port=%d", module, port));
-		args.add(String.format("--%s.endpoints.shutdown.enabled=true", module));
-		launcher.launch(new String[] { module }, args.toArray(new String[args.size()]));
+		args.put("server.port", String.valueOf(port));
+		args.put("endpoints.shutdown.enabled", "true");
+
+		ModuleLaunchRequest moduleLaunchRequest = new ModuleLaunchRequest(module);
```

Comment:
Yeah. Note this was a force alarm, as I forgot to pass args, duh! What we did in XD was to wrap that class with something that remembers the last ports allocated (used as the seed) through a static field. But here we should be fine, as a launch only returns when the module has started